client/src/main/java/org/eclipse/hono/client/Command.java

```diff
@@ -381,10 +418,12 @@ public String toString() {
      * @param deviceId The device id.
      * @return The reply-to-id, starting with the device id.
      */
-    static String getDeviceFacingReplyToId(final String replyToId, final String deviceId) {
-        if (replyToId.startsWith(deviceId + "/")) {
-            return replyToId.replaceFirst(deviceId + "/", deviceId + "/0");
+    public static String getDeviceFacingReplyToId(final String replyToId, final String deviceId) {
+        final boolean replyToContainedDeviceId = replyToId.startsWith(deviceId + "/");
+        final String bitFlagString = encodeReplyToOptions(replyToContainedDeviceId);
+        if (replyToContainedDeviceId) {
+            return replyToId.replaceFirst(deviceId + "/", deviceId + "/" + bitFlagString);
         }
-        return String.format("%s/1%s", deviceId, replyToId);
+        return String.format("%s/%s%s", deviceId, bitFlagString, replyToId);
```

Comment:
it doesn't seem to be necessary to check the *replyToContainedDeviceId* flag, is it? we can always return
`String.format("%s/%s%s", deviceId, bitFlagString, replyToId)`, can't we?

Reply:
If `replyToContainedDeviceId` is true, we would end up with having the device id in there twice, so some string manipulation for the `replyToContainedDeviceId` case is needed anyway (either cutting off the device id or squeezing in the bitFlagString as it is done now).

Reply:
if *replyToId* starts with `$deviceId/` then we return
`$deviceId/$bitFlagString$remainderOfReplyToId`
if *replyToId* **does not** start with `$deviceId/` then $remainderOfReplyToId == $replyToId and we return
`$deviceId/$bitFlagString$replyToId`
which is the same as in the first case, isn't it?

Reply:
I guess you mean it like [this](https://github.com/eclipse/hono/compare/6f59777aebf9e7a2b32f40d198e2d1014a14ccc7..75bdeb227f3ab5d878a7cc2f0ba9c26ce5679217)? That's more concise, yes.